exec/java-exec/src/main/java/org/apache/drill/exec/server/options/FallbackOptionManager.java

```diff
@@ -102,6 +125,29 @@ public void setOption(OptionValue value) {
   }
 
   @Override
+  public void deleteOption(final String name, final OptionType type) {
+    try {
+      SystemOptionManager.getValidator(name); // ensure the option exists
+    } catch (final IllegalArgumentException e) {
+      throw UserException.validationError(e)
+        .build(logger);
```

Comment:
My main issue here is that we have an unfriendly message. One of the things that we should be doing is creating a user friendly message at the point of user exception creation. If you think it was too deep one level deeper, then you should make a friendly user message at this level. Trusting that the system level exception message is meaningful to the user should be avoided.